cmd/iconbridge/chain/near/sender.go

```diff
@@ -88,34 +88,24 @@ func (s *Sender) Segment(ctx context.Context, msg *chain.Message) (tx chain.Rela
 		Receipts: make([][]byte, 0),
 	}
 
-	var msgSize uint64
+	receipt := msg.Receipts[0]
 
-	newMsg = &chain.Message{
-		From:     msg.From,
-		Receipts: msg.Receipts,
+	rlpEvents, err := codec.RLP.MarshalToBytes(receipt.Events)
+	if err != nil {
+		return nil, nil, err
 	}
-	for i, receipt := range msg.Receipts {
-		rlpEvents, err := codec.RLP.MarshalToBytes(receipt.Events)
```

Comment:
Instead of marshaling all events at once, I recommend looping over each event and calculate the size of the final combined message, and only include as many events as allowed by the configured "tx_size_limit".

Reply:
Resolved by https://github.com/icon-project/icon-bridge/pull/545/commits/1915df5fb0d8bdab6d644444c1aa1dc59142f1ac